src/khoj/routers/api_chat.py

```diff
@@ -856,6 +857,30 @@ async def chat(
         _custom_filters.append("site:khoj.dev")
         conversation_commands.append(ConversationCommand.Online)
 
+    conversation = await ConversationAdapters.aget_conversation_by_user(user, conversation_id=conversation_id)
+    if conversation_commands == [ConversationCommand.Summarize]:
+        file_filters = conversation.file_filters
+        if len(file_filters) == 0:
+            llm_response = "No files selected for summarization. Please add files using the section on the left."
+            return StreamingResponse(content=llm_response, media_type="text/event-stream", status_code=200)
+        elif len(file_filters) > 1:
+            llm_response = "Only one file can be selected for summarization."
+            return StreamingResponse(content=llm_response, media_type="text/event-stream", status_code=200)
+        else:
+            try:
+                file_object = await FileObjectAdapters.async_get_file_objects_by_name(user, file_filters[0])
+                contextual_data = " ".join([file.raw_text for file in file_object])
+                summarizeStr = "/" + ConversationCommand.Summarize
+                if q.strip() == summarizeStr:
+                    q = "Create a general summary of the file"
+                response = await extract_relevant_summary(q, contextual_data)
```

Comment:
As is, your output won't stream iteratively. It'll wait until the whole response is generated and then send it in one go. Contrast in the code how `agenerate_chat_response` is used. It's piping the llm response as it comes to the client. 

This is less than ideal, especially because the summarizer can take a bit longer than other queries. But it would require too much refactoring at the moment to get it working, so it's out of scope for this PR. Just wanted to note this so we can improve it later.